### PR TITLE
 Improve aliased from handling in `getReferencedFields`

### PR DESCRIPTION
**Pull request by Page-**
Depends on #59

### CODE REVIEW

File: src/AbstractSQLCompiler.ts [hunk 1]
```diff
@@ -373,64 +373,82 @@ const dataTypeGen = (
...
-					}
+type Scope = _.Dictionary<string>;
+
+const getScope = (rulePart: AbstractSqlQuery, scope: Scope): Scope => {
+	scope = { ...scope };
+	const fromNodes = rulePart.filter(node => node[0] === 'From') as FromNode[];
+	fromNodes.forEach(node => {
+		let nested = node[1];
+		if (nested[0] === 'Alias') {
+			let [, from, alias] = nested;
```
Comment (zvin) [inline src/AbstractSQLCompiler.ts]: This could be a `const`

File: src/AbstractSQLCompiler.ts [hunk 2]
```diff
@@ -373,64 +373,82 @@ const dataTypeGen = (
...
+			if (!_.isString(alias)) {
+				throw new Error('Cannot handle non-string aliases');
+			}
+			switch (from[0]) {
+				case 'Table':
+					scope[alias] = from[1];
+					break;
+				case 'SelectQuery':
+					// Ignore SelectQuery in the From as we'll handle any fields it select
+					// when we recurse in.  With true scope handling however we could prune
```
Comment (zvin) [inline src/AbstractSQLCompiler.ts]: extra space

File: src/AbstractSQLCompiler.ts [hunk 3]
```diff
@@ -373,64 +373,82 @@ const dataTypeGen = (
...
+			let [, from, alias] = nested;
+			if (!_.isString(alias)) {
+				throw new Error('Cannot handle non-string aliases');
+			}
+			switch (from[0]) {
+				case 'Table':
+					scope[alias] = from[1];
+					break;
+				case 'SelectQuery':
+					// Ignore SelectQuery in the From as we'll handle any fields it select
```
Comment (zvin) [inline src/AbstractSQLCompiler.ts]: selects

File: src/AbstractSQLCompiler.ts (hunk 2 above)
Comment (Page-) [reply src/AbstractSQLCompiler.ts]: Yeah it's a habit of an old school typing teacher, it used to be standard to double space after a period :P

**Pull request merged**